### PR TITLE
docs: add typescript typings for plugin authors

### LINKED ISSUE #2: Typescript typings for plugin authors
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@custom-elements-manifest/analyzer@0.1.11` for the project I'm working on.

This should work, the only thing that I'm not sure about is the type imports from the `custom-elements-manifest` package, which didn't appear to work for me, so please check.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@custom-elements-manifest/analyzer/index.d.ts b/node_modules/@custom-elements-manifest/analyzer/index.d.ts
new file mode 100644
index 0000000..994d363
--- /dev/null
+++ b/node_modules/@custom-elements-manifest/analyzer/index.d.ts
@@ -0,0 +1,103 @@
+import * as TS from 'typescript'
+import { Module, Package } from 'custom-elements-manifest';
+
+/** Plugin execution context. Pass arbitrary data here. */
+export type Context = Record<string, unknown>;
+
+export interface CollectPhaseParams {
+  /**
+   * TypeScript API
+   */
+  ts: TS;
+
+  /**
+   * The current TypeScript AST Node
+   */
+  node: TS.Node;
+
+  /**
+   * Plugin execution context. Pass arbitrary data here.
+   */
+  context: Context;
+}
+
+export interface AnalyzePhaseParams {
+  /**
+   * TypeScript API
+   */
+  ts: TS;
+
+  /**
+   * The current TypeScript AST Node
+   */
+  node: TS.Node;
+
+  /**
+   * The current state of the current module's manifest
+   */
+  moduleDoc: Partial<Module>;
+
+  /**
+   * Plugin execution context. Pass arbitrary data here.
+   */
+  context: Context;
+}
+
+export interface ModuleLinkPhaseParams {
+  /**
+   * The completed manifest, i.e. the output of the analyze phase
+   */
+  moduleDoc: Module;
+
+  /**
+   * Plugin execution context. Pass arbitrary data here.
+   */
+  context: Context;
+}
+
+export interface PackageLinkPhaseParams {
+  /**
+   * The completed manifest, i.e. the output of the analyze phase
+   */
+  customElementsManifest: Package;
+
+  /**
+   * Plugin execution context. Pass arbitrary data here.
+   */
+  context: Context;
+}
+
+/**
+ * A Custom Elements Manifest Analyzer plugin
+ */
+export interface Plugin {
+  /**
+   * @summary Plugin hook that runs in the analyze phase.
+   *
+   * Runs for all modules in a project, before continuing to the `analyzePhase`
+   */
+  collectPhase(params: CollectPhaseParams): void;
+
+  /**
+   * @summary Plugin hook that runs in the analyze phase.
+   *
+   * Runs for each AST node in each module.
+   * You can use this phase to access a module's AST nodes and mutate the manifest.
+   */
+  analyzePhase(params: AnalyzePhaseParams): void;
+
+  /**
+   * @summary Plugin hook that runs in the module-link phase.
+   *
+   * Post-processing hook that runs for each module, after analyzing.
+   * All information about your module should now be available.
+   */
+  moduleLinkPhase(params: ModuleLinkPhaseParams): void;
+
+  /**
+   * @summary Plugin hook that runs in the package-link phase.
+   *
+   * Runs once per package, after modules have been parsed and after per-module post-processing
+   */
+  packageLinkPhase(params: PackageLinkPhaseParams): void;
+}
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

### PR DESCRIPTION
**Pull request by bennypowers**
closes #2

### CODE REVIEW

File: packages/analyzer/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import * as TS from 'typescript'
+import { Module, Package } from 'custom-elements-manifest/schema';
```
Comment (thepassle) [inline packages/analyzer/index.d.ts:2]: What about these imports didnt work?

Comment (bennypowers) [reply packages/analyzer/index.d.ts:2]: i forgot the `/schema`. this works now

File: packages/analyzer/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  /**
+   * Plugin execution context. Pass arbitrary data here.
+   */
+  context: Context;
+}
+
+export interface ModuleLinkPhaseParams {
+  /**
+   * The completed manifest, i.e. the output of the analyze phase
```
Comment (thepassle) [inline packages/analyzer/index.d.ts:48]:
```suggestion
   * The completed moduleDoc, i.e. the output of the analyze phase
```

File: packages/analyzer/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+   */
+  context: Context;
+}
+
+/**
+ * A Custom Elements Manifest Analyzer plugin
+ */
+export interface Plugin {
+  /**
+   * @summary Plugin hook that runs in the analyze phase.
```
Comment (thepassle) [inline packages/analyzer/index.d.ts:75]:
```suggestion
   * @summary Plugin hook that runs in the collect phase.
```

File: packages/analyzer/index.d.ts (hunk 2 above)
Comment (bennypowers) [reply packages/analyzer/index.d.ts:48]: committed

**Pull request merged**